### PR TITLE
10.0 Add hr_recruitment_candidate_multi_applicant

### PR DESCRIPTION
**Pull request by leemannd**
Add the possibility to see multiple applications made by one partner

### CODE REVIEW

**Comment by leemannd** [comment]
@gurneyalex Ready for review

File: hr_recruitment_candidate_multi_applicant/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+========================
+candiate multi applicant
```
Comment (gurneyalex) [inline hr_recruitment_candidate_multi_applicant/README.rst]: s/candiate/candidate/

File: hr_recruitment_candidate_multi_applicant/README.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+========================
+
+Helper module to detect multiple applications to jobs from the same candidate.
```
Comment (gurneyalex) [inline hr_recruitment_candidate_multi_applicant/README.rst:9]: explain that "same candidate is detected by same mail or same phone"

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @api.depends('email_from', 'partner_phone')
+    def _compute_same_candidate_application_count(self):
+        for applicant in self:
+            domain = ['|', ('active', '=', False), ('active', '=', True),
+                      '|', '&', ('email_from', '=', applicant.email_from),
+                      ('email_from', '!=', False), '&',
+                      ('partner_phone', '=', applicant.partner_phone),
+                      ('partner_phone', '!=', False)]
+            if not isinstance(applicant.id, models.NewId):
+                domain = [('id', '!=', applicant.id)] + domain
```
Comment (gurneyalex) [inline hr_recruitment_candidate_multi_applicant/models/hr_applicant.py:32]: probably the domain building should be extracted in a helper method to make it easier to extend the module.

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            same_app = self.search(domain)
```
Comment (gurneyalex) [inline hr_recruitment_candidate_multi_applicant/models/hr_applicant.py]: use  `active_test = False`in the search and remove the active = True or active = False in the domain.

**Review by gurneyalex — CHANGES_REQUESTED** [review]
a few changes suggested

**Comment by leemannd** [comment]
@gurneyalex Updated. Thanks for the feedback. To be squashed before merge.

**Review by gurneyalex — APPROVED** [review]

**Comment by leemannd** [comment]
Squashed

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py (hunk 3 above)
Comment (leemannd) [reply hr_recruitment_candidate_multi_applicant/models/hr_applicant.py:32]: @feketemihai I'm putting me a reminder to do it this week or early the next one

**Review by feketemihai — APPROVED** [review]
The extraction of domain was good, but not a blocker.

File: hr_recruitment_candidate_multi_applicant/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+from . import models
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/__init__.py:2]: Missing `-*- coding: utf-8 -*-` : https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

File: hr_recruitment_candidate_multi_applicant/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1 @@
+from . import hr_applicant
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/models/__init__.py:2]: Missing -*- coding: utf-8 -*- : https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Author: Denis Leemann
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/models/hr_applicant.py]: Alphabetically ordered: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    @api.depends('email_from', 'partner_phone')
+    def _compute_same_candidate_application_count(self):
+        for applicant in self:
+            domain = ['|', '&', ('email_from', '=', applicant.email_from),
+                      ('email_from', '!=', False), '&',
+                      ('partner_phone', '=', applicant.partner_phone),
+                      ('partner_phone', '!=', False)]
+            if not isinstance(applicant.id, models.NewId):
+                domain = [('id', '!=', applicant.id)] + domain
+            same_app = self.with_context(active_test=False).search(domain)
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/models/hr_applicant.py]: variable name should be `same_apps`

File: hr_recruitment_candidate_multi_applicant/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,2 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from . import test_multi_applicant
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/tests/__init__.py:3]: Missing -- coding: utf-8 -- : https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

File: hr_recruitment_candidate_multi_applicant/tests/test_multi_applicant.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+                'name': 'John Doe',
+                'email_from': 'jdoe@camptocamp.com',
+            })
+
+        def test_compute_same_email(self):
+            self.assertEqual(1, self.toto.same_candidate_application_count)
+            self.assertEqual(1, self.toto2.same_candidate_application_count)
+            self.assertEqual(0, self.jdoe.same_candidate_application_count)
+
+        def test_same_phone(self):
```
Comment (hieulucky111) [inline hr_recruitment_candidate_multi_applicant/tests/test_multi_applicant.py:31]: Same with `test_compute_same_email`?

**Review by hieulucky111 — APPROVED** [review]
Some minor adjustments, LGTM over all

File: hr_recruitment_candidate_multi_applicant/tests/test_multi_applicant.py (hunk 10 above)
Comment (leemannd) [reply hr_recruitment_candidate_multi_applicant/tests/test_multi_applicant.py:31]: @hieulucky111 I don't undersand your point. The email is tested in => https://github.com/OCA/hr/pull/373/files/fa410c9efb6192e0ce51a251679d122e6d61e033#diff-59fe9f77b440f08b6354e3e4b21b3720R26

File: hr_recruitment_candidate_multi_applicant/views/applicant.xml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+  <record id="crm_case_form_view_job_specific" model="ir.ui.view">
+    <field name="name">view_project_form_specific</field>
+    <field name="model">hr.applicant</field>
+    <field name="inherit_id" ref="hr_recruitment.crm_case_form_view_job"/>
+    <field name="arch" type="xml">
+      <xpath expr="//div[@class='oe_button_box']" position="inside">
```
Comment (nikul-serpentcs) [inline hr_recruitment_candidate_multi_applicant/views/applicant.xml]: here use `hasclass`

**Review by nikul-serpentcs — COMMENTED** [review]
Small Change

File: hr_recruitment_candidate_multi_applicant/views/applicant.xml [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+        <button name="action_view_applicants" class="oe_stat_button" type="object" icon="fa-users"
```
Comment (nikul-serpentcs) [inline hr_recruitment_candidate_multi_applicant/views/applicant.xml:9]: Remove `icon` and add `<i class="fa-users" />`

**Comment by leemannd** [comment]
Closing my work that is 1 year old

**Comment by leemannd** [comment]
Closing my work that is 1 year old

File: hr_recruitment_candidate_multi_applicant/models/hr_applicant.py (hunk 3 above)
Comment (leemannd) [reply hr_recruitment_candidate_multi_applicant/models/hr_applicant.py:32]: I'm wondering, as it's old. Can we get it merge and improve it if needed?

**Comment by leemannd** [comment]
@rafaelbn As you are at a CS, can you take a look at it?
Maybe we want to get it merged?

If there are changes to be done, please amend my PR or feel free to close it.

**Review by rafaelbn — APPROVED** [review]
I'm not PSC of HR but this PR should be merged

**Comment by rafaelbn** [comment]
@gurneyalex could merge this

**Comment by nikul-serpentcs** [comment]
@gurneyalex @rafaelbn squashing commits are remaining :smile_cat:

**Pull request merged**